server/block/note.go

```diff
@@ -48,13 +49,40 @@ func (n Note) EncodeNBT() map[string]any {
 
 // Activate ...
 func (n Note) Activate(pos cube.Pos, _ cube.Face, tx *world.Tx, _ item.User, _ *item.UseContext) bool {
+	n.trigger(pos, tx)
+	return true
+}
+
+// RedstoneUpdate ...
+func (n Note) RedstoneUpdate(pos cube.Pos, tx *world.Tx) {
+	if n.powered(pos, tx) {
+		n.trigger(pos, tx)
+	}
+}
+
+// trigger plays the note if there's space above the note block.
+func (n Note) trigger(pos cube.Pos, tx *world.Tx) {
+	// Can only tune if there's air above
 	if _, ok := tx.Block(pos.Side(cube.FaceUp)).(Air); !ok {
-		return false
+		return
 	}
 	n.Pitch = (n.Pitch + 1) % 25
 	n.playNote(pos, tx)
 	tx.SetBlock(pos, n, &world.SetOpts{DisableBlockUpdates: true, DisableLiquidDisplacement: true})
-	return true
+}
+
+// powered checks if the note block is receiving redstone power.
+func (n Note) powered(pos cube.Pos, tx *world.Tx) bool {
+	for _, face := range cube.Faces() {
+		adjacentPos := pos.Side(face)
+		if power := tx.RedstonePower(adjacentPos, face.Opposite(), true); power > 0 {
+			return true
+		}
+		if power := tx.RedstonePower(adjacentPos, face.Opposite(), false); power > 0 {
+			return true
+		}
+	}
+	return false
 }
```

Comment:
Might be a good idea to have a generic version of this? Lots of blocks, e.g. redstone lamps, TNT, dispensers need this exact same logic

Doors can also use this generic method, e.g. `powered(bottomPos, tx) || powered(topPos, tx)`

Reply:
Also why are we checking redstone power without account for dust here?